addons/product/static/src/product_catalog/search/search_panel.js

```diff
@@ -1,54 +1,184 @@
+                newTag.set('count', count);
+                sections.set(name, newTag);
             }
         });
 
         return sections;
     }
 
-    toggleSectionFilterValue(filterId, attrIds, { currentTarget }) {
-        attrIds.forEach(id => {
+    toggleTagFilterValue(filterId, tagIds, { currentTarget }) {
+        tagIds.forEach(id => {
             this.toggleFilterValue(filterId, id, { currentTarget });
         })
     }
+
+    enableSectionInput() {
+        this.state.isAddingSection = true;
+        setTimeout(() => document.querySelector('.o_section_input')?.focus(), 100);
+    }
+
+    onDragStart(sectionId, ev) {
+        ev.dataTransfer.setData("text/plain", sectionId);
+    }
+
+    onDragOver(ev) {
+        ev.preventDefault();
+    }
+
+    onDrop(targetSecId, ev) {
+        ev.preventDefault();
+        const moveSecId = ev.dataTransfer.getData("text/plain");
+        if (moveSecId !== targetSecId) this.reorderSections(moveSecId, targetSecId);
+    }
+
+    onSectionInputKeydown(event) {
+        if (event.key === "Enter") {
+            this.createSection();
+        } else if (event.key === "Escape") {
+            this.state.isAddingSection = false;
+            this.state.newSectionName = "";
+        }
+    }
+
+    setSelectedSection(sectionId=null, filtered=false) {
+        this.env.searchModel.setSelectedSection(sectionId, filtered);
+    }
+
+    async createSection() {
+        const sectionName = this.state.newSectionName.trim();
+        if (!sectionName) return this.state.isAddingSection = false;
+
+        const section = await rpc("/product/catalog/create_section", {
+            res_model: this.env.model.config.context.product_catalog_order_model,
+            order_id: this.env.model.config.context.order_id,
+            section_name: sectionName,
+        });
+
+        if (section?.id) {
+            this.state.sectionOfSections.set(section.id, {
+                name: section.name,
+                sequence: section.sequence,
+                line_count: 0,
+            });
+            this.setSelectedSection(section.id);
+        }
+        this.state.isAddingSection = false;
+        this.state.newSectionName = "";
+    }
+
+    async loadSections() {
+        if (!this.isNonMrpOrderActive) return;
+        const sections = await rpc(
+            "/product/catalog/get_sections", {
+            res_model: this.env.model.config.context.product_catalog_order_model,
+            order_id: this.env.model.config.context.order_id,
+            child_field: this.env.model.config.context.child_field,
+        });
+
+        const sectionMap = this.state.sectionOfSections || new Map();
+        for (const {id, name, sequence, line_count} of sections) {
+            const existingSection = sectionMap.get(id);
+            if (!existingSection) {
+                sectionMap.set(id, {name, sequence, line_count});
+            }
+        }
+        this.state.sectionOfSections = sectionMap;
+    }
```

Comment:
```suggestion
    _getSectionRPCPayload(extra = {}) {
        const ctx = this.env.model.config.context;
        return {
            res_model: ctx.product_catalog_order_model,
            order_id: ctx.order_id,
            child_field: ctx.child_field,
            ...extra,
        };
    async loadSections() {
        if (!this.isNonMrpOrderActive) return;
        const sections = await rpc(
            "/product/catalog/get_sections", this._getSectionRPCPayload());

        const sectionMap = this.state.sectionOfSections || new Map();
        for (const {id, name, sequence, line_count} of sections) {
            const existingSection = sectionMap.get(id);
            if (!existingSection) {
                sectionMap.set(id, {name, sequence, line_count});
            }
        }
        this.state.sectionOfSections = sectionMap;
    }
```

You can try making a single function and use it for `createSection`, `loadSections` and `reorderSections`